docs/upgrading_to_v16.0.md

```diff
@@ -0,0 +1,28 @@
+# Upgrading to v16.0
+
+The v16.0 release of *kubernetes-engine* is a backwards incompatible release.
+
+### cluster_autoscaling modified
+The `cluster_autoscaling` variable has been modified to require a `gpu_resources` value.
+
+```diff
+ module "gke" {
+   source  = "terraform-google-modules/kubernetes-engine/google//modules/private-cluster"
+-  version = "~> 15.0"
++  version = "~> 16.0"
+
+  cluster_autoscaling = {
+    enabled             = true
+    autoscaling_profile = "BALANCED"
+    min_cpu_cores       = 1
+    max_cpu_cores       = 100
+    min_memory_gb       = 1
+    max_memory_gb       = 1000
++   gpu_resources = [{
++     resource_type = "nvidia-tesla-t4"
++     minimum       = 1
++     maximum       = 3
++   }]
```

Comment:
@JamesTimms we should showcase how to preserve historical behavior([example](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/docs/upgrading_to_v10.0.md#default-machine-type))
```suggestion
+   gpu_resources = []
```